allauth/socialaccount/providers/hubspot/provider.py

```diff
@@ -0,0 +1,29 @@
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class Scope(object):
+    oauth2 = 'oauth'
+
+
+class HubSpotAccount(ProviderAccount):
+    pass
+
+
+class HubSpotProvider(OAuth2Provider):
+    id = 'hubspot'
+    name = 'HubSpot'
+    account_class = HubSpotAccount
+
+    def extract_common_fields(self, data):
+        return dict(username=data.get('user'))
+
+    def extract_uid(self, data):
+        return str(data['id'])
```

Comment:
I checked the API documentation:

https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-token-information

```
{
  "token": "CJSP5qf1KhICAQEYs-gDIIGOBii1hQIyGQAf3xBKmlwHjX7OIpuIFEavB2-qYAGQsF4",
  "user": "test@hubspot.com",
  "hub_domain": "demo.hubapi.com",
  "scopes": [
    "contacts",
    "automation",
    "oauth"
  ],
  "hub_id": 62515,
  "app_id": 456,
  "expires_in": 21588,
  "user_id": 123,
  "token_type": "access"
}
```

There is no mention of an `id` key there.